### PR TITLE
Feature load color config

### PR DESCRIPTION
**Pull request by mel-flygare96**
Cards this pull request addresses:
Vision -- Load fitted dataset settings from config file

To test:

Pull feature_load_color_config
Checkout master
Merge feature_load_color_config in
Catkin make
run roslaunch op3_ball_detector op3_ball_detector.launch

If you review this PR, please add your name and email below.

Reviewers:
Reviewed-by: Brady Testa <btesta@pdx.edu>
Reviewed-by: Alice Michael <malice@pdx.edu>
Review-by: Nicholas Gilbert <gnick@pdx.edu>

### CODE REVIEW

**Review by BradyT98 — APPROVED** [review]
Looks good. I don't see anything concerning,

File: op3_ball_detector/include/op3_ball_detector/ball_detector.h [hunk 1]
```diff
@@ -84,10 +85,15 @@ class BallDetector
...
   bool saveImageCallback(op3_ball_detector::SaveImage::Request &req, op3_ball_detector::SaveImage::Response &res);
+  bool switchDetectionCallback(op3_ball_detector::SwitchDetection::Request &req, op3_ball_detector::SwitchDetection::Response &res);
+
+  bool loadDetectionSettings();
+  void applyDetectionSettings(); // types TBD
 
   void resetParameter();
   void publishParam();
 
+
```
Comment (Sappytomb796) [inline op3_ball_detector/include/op3_ball_detector/ball_detector.h:96]: This line should be removed if we can squish that in with whichever commit added it, if not it's ok with me to merge

File: op3_ball_detector/src/ball_detector.cpp [hunk 2]
```diff
@@ -69,6 +73,12 @@ BallDetector::BallDetector()
   nh_.param<int>("ellipse_size", detect_config.ellipse_size, params_config_.ellipse_size);
   nh_.param<bool>("filter_debug", detect_config.debug, params_config_.debug);
 
+  // nh_.param<std::string>("name", color_config.name, params_color_.name);
+  // nh_.param<int>("test_val", color_config.test_val, params_color_.test_val);
```
Comment (Sappytomb796) [inline op3_ball_detector/src/ball_detector.cpp]: What is the reason for commenting out these lines?  If we are submitting a pull request, we should be making sure to mark to take dead code out later if we want to merge it to master.

File: op3_ball_detector/srv/SwitchDetection.srv [hunk 3]
```diff
@@ -0,0 +1,3 @@
+
```
Comment (Sappytomb796) [inline op3_ball_detector/srv/SwitchDetection.srv:1]: What is up with the empty line?

Comment (gilben1) [reply op3_ball_detector/srv/SwitchDetection.srv:1]: Necessary formatting

**Review by Sappytomb796 — APPROVED** [review]

**Review by gilben1 — APPROVED** [review]
Despite the empty line, all of the changes look sane.

**Pull request merged**